DRdigi/src/DigiSiPM.cpp

```diff
@@ -63,22 +61,22 @@ StatusCode DigiSiPM::execute() {
     auto digiHit = digiHits->create();
     auto waveform = waveforms->create();
 
-    sipm::SiPMAnalogSignal anaSignal = m_sensor->signal();
-    sipm::SiPMDigitalSignal digiSignal = m_adc->digitize(anaSignal);
+    // Using only analog signal (ADC conversion is still experimental)
+    const sipm::SiPMAnalogSignal anaSignal = m_sensor->signal();
 
-    int integral = digiSignal.integral(m_gateStart,m_gateL,m_thres);   // (intStart, intGate, threshold)
-    double toa = digiSignal.toa(m_gateStart,m_gateL,m_thres);          // (intStart, intGate, threshold)
+    const double integral = anaSignal.integral(m_gateStart,m_gateL,m_thres);   // (intStart, intGate, threshold)
+    const double toa = anaSignal.toa(m_gateStart,m_gateL,m_thres);          // (intStart, intGate, threshold)
 
     digiHit.setAmplitude( integral );
     digiHit.setCellID( rawhit.getCellID() );
-    digiHit.setTimeStamp( static_cast<int>((toa+m_gateStart)/m_sampling) );
+    // Toa and m_gateStart are in ns
+    digiHit.setTimeStamp( static_cast<int>(toa+m_gateStart) );
```

Comment:
Actually, the definition for `timeStamp` here is not the time itself (in ns) but the time in the unit of the sampling time - otherwise we'll lose accuracy below 1 ns since `edm4hep::RawCalorimeterHit` saves the `timeStamp` as an integer. The ToA will be saved correctly in the final product `edm4hep::CalorimeterHit` in ns [here](https://github.com/HEP-FCC/dual-readout/blob/158c6f6714294f50d6d02c441b9f9b70247325d6/DRreco/src/DRcalib2D.cpp#L59).